model/project_ref.go

```diff
@@ -133,6 +134,71 @@ type ProjectRef struct {
 	// Filter/view settings
 	ProjectHealthView ProjectHealthView `bson:"project_health_view" json:"project_health_view" yaml:"project_health_view"`
 	ParsleyFilters    []parsley.Filter  `bson:"parsley_filters,omitempty" json:"parsley_filters,omitempty"`
+
+	// GitHubDynamicTokenPermissionGroups is a list of permission groups for GitHub dynamic access tokens.
+	GitHubDynamicTokenPermissionGroups GitHubDynamicTokenPermissionGroups `bson:"github_dynamic_token_permission_groups,omitempty" json:"github_dynamic_token_permission_groups,omitempty" yaml:"github_dynamic_token_permission_groups,omitempty"`
+
+	// GitHubPermissionGroupByRequester is a mapping of requester type to the user defined GitHub permission groups above.
+	GitHubPermissionGroupByRequester map[string]string `bson:"github_token_permission_by_requester,omitempty" json:"github_token_permission_by_requester,omitempty" yaml:"github_token_permission_by_requester,omitempty"`
+}
+
+// GitHubDynamicTokenPermissionGroup is a permission group for GitHub dynamic access tokens.
+type GitHubDynamicTokenPermissionGroup struct {
+	Name string `bson:"name,omitempty" json:"name,omitempty" yaml:"name,omitempty"`
+	// Permissions are a key-value pair of GitHub token permissions to their permission level
+	Permissions github.InstallationPermissions `bson:"permissions,omitempty" json:"permissions,omitempty" yaml:"permissions,omitempty"`
+}
+
+// GitHubDynamicTokenPermissionGroups is used to specify groups of permissions and requesters for
+// GitHub dynamic access tokens.
+type GitHubDynamicTokenPermissionGroups []GitHubDynamicTokenPermissionGroup
+
+var defaultGitHubTokenPermissionGroup = GitHubDynamicTokenPermissionGroup{}
+
+// GetGitHubPermissionGroup returns the GitHubDynamicTokenPermissionGroup for the given requester.
+// If the requester is not found, it returns the default.
+func (p *ProjectRef) GetGitHubPermissionGroup(requester string) GitHubDynamicTokenPermissionGroup {
+	if p.GitHubPermissionGroupByRequester == nil {
+		return defaultGitHubTokenPermissionGroup
```

Comment:
That makes sense!

Reply:
I kept the default but I added a new field that is tested at the api service layer called 'NoPermissions'. If a group is decoded with no values filled in, it will have it set to true. Lmk what you think!